File: fsharp/grains/Grains.fs
module Grains

    let square (n:int):Result<uint64,string> =
        match n > 0 && n < 65 with
        | true -> Ok(pown 2UL (n - 1))
        | false -> Error("square must be between 1 and 64")

    let total:Result<uint64,string> =
        seq{0..63}
        |> Seq.sumBy (fun (n:int) -> pown 2UL n)
        |> Ok

File: solutions/scala/grains/1/src/main/scala/Grains.scala
object Grains
{
    def square(n:Int):Option[BigInt] =
        n > 0 && n < 65 match
        {
            case true => Some(BigInt(2).pow(n - 1))
            case false => None
        }

    def total:BigInt = Seq.range(0, 64).map((n:Int) => BigInt(2).pow(n)).sum
}